lib/ore/aes128.rb

```diff
@@ -71,6 +71,13 @@ def encrypt(plaintext)
         encrypt_bool(plaintext)
       when Range
         encrypt_range(plaintext)
+      when Date
+        coerced = (plaintext - Date.new(1970, 1, 1)).to_i
+        encrypt_u64(coerced)
+      when Time
+        # Millis since the epoch
+        coerced = (plaintext.to_r * 1000).to_i
+        encrypt_u64(coerced)
```

Comment:
By storing &lt;anything>-since-the-epoch in a u64, you're breaking the ability to store dates/times before 1970.  Might not be a problem in many cases, but anyone who's storing dates of birth, for instance, are probably going to need to handle people older than about 52.

Also, I'd be a lot more comfortable if we did as much of this encoding work as possible in Rust, not so much because the chances that two different languages have all date/time handling implemented identically is... minimal, but because it'll reduce the number of times this conversation has to happen as we implement in more languages.

Reply:
Why is seconds since the epoch limited to 52 years?